ext/spl/spl_array.c

```diff
@@ -1888,6 +1897,19 @@ SPL_METHOD(Array, __unserialize)
 	}
 
 	object_properties_load(&intern->std, Z_ARRVAL_P(members_zv));
+
+	if (iterator_class_zv && Z_TYPE_P(iterator_class_zv) == IS_STRING) {
+		zend_class_entry *ce = zend_lookup_class(Z_STR_P(iterator_class_zv));
+
+		if (ce) {
+			intern->ce_get_iterator = ce;
+		} else {
+			zend_throw_exception_ex(spl_ce_UnexpectedValueException, 0,
```

Comment:
Is there any more appropriate exception class to use than `UnexpectedValueException`?

Reply:
Given that the other error condition already uses it, UnexpectedValueException is fine I think.